zim/notebook/notebook.py

```diff
@@ -835,11 +835,11 @@ def _update_link_tag(self, elt, source, target, oldhref):
 
 		if elt.gettext() == elt.get('href'):
 			elt[:] = [link]
-		elif elt.gettext() == oldhref.parts()[-1] and len(elt) == 1:
+		elif oldhref.parts() and elt.gettext() == oldhref.parts()[-1] and len(elt) == 1:
 			# We are using short links and the link text was short link
 			# and there were no sub-node (like bold text) that would be cancelled.
 			# Related to 'short_links' but not checking the property here.
-			short = newhref.parts()[-1]
+			short = newhref.parts()[-1] if newhref.parts() else ''
```

Comment:
This makes pasting the heading produce a link like `#this` instead of `this` like all the other headings do. Imo they should all behave the same way; if that way is `this`, it can be achieved with `elif oldhref.parts() and newhref.parts() and ...` on line 838